pkg/syncer/status/status_process_test.go

```diff
@@ -750,3 +751,86 @@ func (f *fakeSyncerInformers) SyncableGVRs() (map[schema.GroupVersionResource]*r
+		{"annotation", args{unstrob: &unstructured.Unstructured{Object: map[string]interface{}{"metadata": map[string]interface{}{"annotations": map[string]interface{}{storage.DelayStatusSyncing: "true"}}}}}, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := hasDelayStatusSyncingAnnotation(tt.args.unstrob); got != tt.want {
+				t.Errorf("hasDelayStatusSyncingAnnotation() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_hasDelayStatusSyncingAnnotationChanged(t *testing.T) {
+	type args struct {
+		oldUnstrob *unstructured.Unstructured
+		newUnstrob *unstructured.Unstructured
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{"no annotation", args{oldUnstrob: &unstructured.Unstructured{}, newUnstrob: &unstructured.Unstructured{}}, false},
+		{"annotation but incorrect content", args{oldUnstrob: &unstructured.Unstructured{Object: map[string]interface{}{"metadata": map[string]interface{}{"annotations": map[string]interface{}{storage.DelayStatusSyncing: "foo"}}}}, newUnstrob: &unstructured.Unstructured{Object: map[string]interface{}{"metadata": map[string]interface{}{"annotations": map[string]interface{}{storage.DelayStatusSyncing: "foo"}}}}}, false},
+		{"annotation is still there and hasn't changed", args{oldUnstrob: &unstructured.Unstructured{Object: map[string]interface{}{"metadata": map[string]interface{}{"annotations": map[string]interface{}{storage.DelayStatusSyncing: "true"}}}}, newUnstrob: &unstructured.Unstructured{Object: map[string]interface{}{"metadata": map[string]interface{}{"annotations": map[string]interface{}{storage.DelayStatusSyncing: "true"}}}}}, true},
+		{"annotation was removed", args{oldUnstrob: &unstructured.Unstructured{Object: map[string]interface{}{"metadata": map[string]interface{}{"annotations": map[string]interface{}{storage.DelayStatusSyncing: "true"}}}}, newUnstrob: &unstructured.Unstructured{Object: map[string]interface{}{"metadata": map[string]interface{}{"annotations": map[string]interface{}{"foo": "true"}}}}}, false},
+		{"annotation was added", args{oldUnstrob: &unstructured.Unstructured{Object: map[string]interface{}{"metadata": map[string]interface{}{"annotations": map[string]interface{}{"foo": "true"}}}}, newUnstrob: &unstructured.Unstructured{Object: map[string]interface{}{"metadata": map[string]interface{}{"annotations": map[string]interface{}{storage.DelayStatusSyncing: "true"}}}}}, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := hasDelayStatusSyncingAnnotationChanged(tt.args.oldUnstrob, tt.args.newUnstrob); got != tt.want {
+				t.Errorf("hasDelayStatusSyncingAnnotationChanged() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+// Both called functions have good unit tests already so this is only testing the OR case
+func Test_notDeepEqualFinalizersAndStatusOrDelayStatusSyncingAnnotationChanged(t *testing.T) {
+	t.Run("true / true", func(t *testing.T) {
```

Comment:
nit: test names and logging output is best written for folks that accidentally break these but have no idea how the system works - names like `"true / true"` and the `t.Errorf()` output in this test are opaque and fairly arcane, let's consider how we might make them more clear.

Reply:
Well it's a "simple" OR case so I didn't think too much 😅 , I'll try to think of something better.